authorization/token/repository/token_blackbox_test.go

```diff
@@ -213,3 +214,42 @@ func (s *tokenBlackBoxTest) TestSetStatusFlagsForIdentity() {
 	t4Loaded := s.Graph.LoadToken(t4.TokenID())
 	require.True(s.T(), t4Loaded.Token().Valid())
 }
+
+func (s *tokenBlackBoxTest) TestCleanupExpiredTokens() {
+	now := time.Now()
+	yesterday := now.AddDate(0, 0, -1)
+	tomorrow := now.AddDate(0, 0, 1)
+
+	s.Graph.CreateToken(yesterday)
+	s.Graph.CreateToken(yesterday)
+	s.Graph.CreateToken(now)
+	s.Graph.CreateToken(now)
+	s.Graph.CreateToken(now)
+	s.Graph.CreateToken(tomorrow)
+	s.Graph.CreateToken(tomorrow)
+
+	require.Equal(s.T(), s.countTokens(s.T()), 7)
+
+	// Let's start by cleaning up all tokens that expired more than 1 hour ago
+	err := s.repo.CleanupExpiredTokens(s.Ctx, 1)
+	require.NoError(s.T(), err)
+
+	// We should be left with 5 tokens (the "yesterday" tokens should now be gone)
+	require.Equal(s.T(), s.countTokens(s.T()), 5)
+
+	// Now let's clean up all the expired tokens, without any retention
+	err = s.repo.CleanupExpiredTokens(s.Ctx, 0)
+	require.NoError(s.T(), err)
+
+	// We should now be left with just 2 tokens
+	require.Equal(s.T(), s.countTokens(s.T()), 2)
```

Comment:
Let's check the specific tokens. Not the number but their IDs to make sure the expected tokens are deleted.

Reply:
Done